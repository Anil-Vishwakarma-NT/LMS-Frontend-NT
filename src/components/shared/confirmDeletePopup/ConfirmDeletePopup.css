.confirm-modal-overlay {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, 0.5);
    display: flex;
    z-index: 2000;
    justify-content: center;
    align-items: center;
  }
  
  .confirm-modal-container {
    background-color: white;
    padding: 20px;
    border-radius: 8px;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
    max-width: 280px;
    width: 100%;
  }
  
  .confirm-modal-header {
    font-size: 1.5rem;
    margin-bottom: 15px;
    font-family: "Poppins", sans-serif;
  }
  
  .confirm-modal-message {
    margin-bottom: 20px;
    font-size: 0.9rem;
    font-family: "Poppins", sans-serif;
  }
  
  .confirm-modal-buttons {
    display: flex;
    justify-content: center;
    gap: 3rem;
  }
  
  .button {
    padding: 10px 20px;
    border: none;
    border-radius: 4px;
    cursor: pointer;
    font-weight: bold;
  }
  
  .btn-cancel {
    background-color: #ccc;
  }
  
  .btn-confirm {
    background-color: #ff4d4d;
    color: white;
  }

  .btn-confirm:hover{
    background-color: #e21c1c;
    color: #fff;
  }
  