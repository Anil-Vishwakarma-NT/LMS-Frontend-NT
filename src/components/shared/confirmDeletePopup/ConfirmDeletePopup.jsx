import React from 'react';
import './ConfirmDeletePopup.css';

const ConfirmDeletePopup = ({ isOpen, onClose, onConfirm }) => {
  if (!isOpen) return null;

  return (
    <div className="confirm-modal-overlay">
      <div className="confirm-modal-container">
        <h3 className="confirm-modal-header">Confirm Delete</h3>
        <p className="confirm-modal-message">Are you sure you want to delete this item?</p>
        <div className="confirm-modal-buttons">
          <button className="button btn-cancel" onClick={onClose}>Cancel</button>
          <button className="button btn-confirm" onClick={onConfirm}>Delete</button>
        </div>
      </div>
    </div>
  );
};

export default ConfirmDeletePopup;
