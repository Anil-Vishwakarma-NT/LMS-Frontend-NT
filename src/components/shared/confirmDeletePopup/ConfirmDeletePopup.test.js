import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import ConfirmDeletePopup from './ConfirmDeletePopup';

test('does not render the popup when isOpen is false', () => {
  const { container } = render(
    <ConfirmDeletePopup isOpen={false} onClose={() => {}} onConfirm={() => {}} />
  );
  expect(container).toBeEmptyDOMElement(); 
});  

test('calls onClose when Cancel button is clicked', () => {
  const onCloseMock = jest.fn();
  
  render(
    <ConfirmDeletePopup isOpen={true} onClose={onCloseMock} onConfirm={() => {}} />
  );
  
  const cancelButton = screen.getByText(/Cancel/i);
  fireEvent.click(cancelButton);
  
  expect(onCloseMock).toHaveBeenCalled(); 
});


test('matches snapshot when isOpen is true', () => {
  const { asFragment } = render(
    <ConfirmDeletePopup isOpen={true} onClose={() => {}} onConfirm={() => {}} />
  );
  
  expect(asFragment()).toMatchSnapshot(); 
});
