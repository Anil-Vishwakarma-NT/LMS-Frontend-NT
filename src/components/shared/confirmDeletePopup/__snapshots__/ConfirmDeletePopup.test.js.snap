// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when isOpen is true 1`] = `
<DocumentFragment>
  <div
    class="confirm-modal-overlay"
  >
    <div
      class="confirm-modal-container"
    >
      <h3
        class="confirm-modal-header"
      >
        Confirm Delete
      </h3>
      <p
        class="confirm-modal-message"
      >
        Are you sure you want to delete this item?
      </p>
      <div
        class="confirm-modal-buttons"
      >
        <button
          class="button btn-cancel"
        >
          Cancel
        </button>
        <button
          class="button btn-confirm"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
